### ISSUE TITLE
axe core 4.1.1 issues in tree view

### ISSUE THREAD

**Issue by sumegha26** [vmware/clarity#5408]
## Describe the bug

Two issues reported by Chrome axe plugin on accessibility of clarity tree view 

## How to reproduce

1.  Go to  https://clarity.design/documentation/tree-view
2.  Click on checkbox-tree - https://clarity.design/documentation/tree-view#checkbox-tree
3.  Right click inspect and run axe plugin in chrome
4.  See two axe issues -

a) Certain ARIA roles must be contained by particular parents
b) Certain ARIA roles must contain particular children

## Expected behavior

These accessibility issues should be resolved

- For first -  `div` under `clr-tree-node`  has role `treeitem` also immediate parent has role `treeitem` . According axe core

Required ARIA parents role not present: `group`, `tree`

- For second - `clr-tree` has role `tree` and  immediate child `clr-recursive-children` has role `group` .  According axe core

Required ARIA child role not present: `treeitem`

## Additional notes

- Following the order of role for nested tree items should fix this issue

**Comment by gnomeontherun** [comment]
This should likely be resolved by #6020 by @jeeyun.

**Comment by jeeyun** [comment]
Just some notes on this:
- It seems that we have a minor bug in the Angular version where both the `clr-tree-node` and its child `div` have role of `treeitem` in this case (haven't dug into check if this is always the case or only under certain scenario)
-If you wanted to try running the axe tool we'd now have to use the angular version of the doc (https://v4.clarity.design/documentation/tree-view#checkbox-tree) since we don't currently have any rendered component in the url mentioned in the ticket
- This particular bug won't be present in the core tree view as @gnomeontherun mentioned. However, axe tool seems a bit limited / incorrect in the report it generates. I tried running it against our WIP core tree view and got a similar bug report even though the a11y tree generated by chrome looks good, and the parent element it mentions is present. 

Screenshot of axe on core tree view:
![Screen Shot 2021-06-16 at 12 19 36 PM](https://user-images.githubusercontent.com/1827742/122281888-a51a0580-ce9f-11eb-9579-c7d2d736fd4b.png)

DOM and a11y tree generated by chrome:
![Screen Shot 2021-06-16 at 12 20 40 PM](https://user-images.githubusercontent.com/1827742/122281921-acd9aa00-ce9f-11eb-93b2-077b7d829154.png)

It may be that axe doesn't know how to evaluate slotted elements correctly in terms of scanning the DOM to generate their report. We worked with a11y team to ensure the tree is compliant and navigable with screen reader.

**Comment by gnomeontherun** [comment]
Our goal is to pass manual validation with our professional accessibility team and take their guidance over automated tools. I'm going to make this as not an issue with Core.
